### PR TITLE
feat(conditions): allow timezone in current-time/schedule/timestamp

### PR DESCRIPTION
**Pull request by sileht**
This change allow to add timezone in format [Europe/Paris] to some time
based condition value.

Change-Id: I00612114bdaf74b4ef41151d8fa4bc4b0ad11373

### CODE REVIEW

File: docs/source/configuration.rst [hunk 1]
```diff
@@ -188,6 +188,11 @@ Examples
         actions:
           close:
             message: It's too late for this!
+      - name: merge on working hour
+        conditions:
+          - schedule: MON-FRI 09:00-19:00[America/Vancouver]
```
Comment (jd) [inline docs/source/configuration.rst:193]:
```suggestion
          - schedule: Mon-Fri 09:00-19:00[America/Vancouver]
```

File: releasenotes/notes/timezone-79b0d5e97fed6bed.yaml [hunk 2]
```diff
@@ -0,0 +1,3 @@
+---
+fixes:
+  - allow to use timezone in current-time/schedule/isoformat conditions
```
Comment (jd) [inline releasenotes/notes/timezone-79b0d5e97fed6bed.yaml:3]:
```suggestion
features:
  - Allow to specify timezone in ``current-time``, ``schedule`` and ??? conditions
```
what is isoformat?

File: mergify_engine/rules/parser.py [hunk 3]
```diff
@@ -153,13 +165,27 @@ def _iso_datetime(tokens: typing.List[pyparsing.Token]) -> datetime.datetime:
...
 )
-_time_range = (_match_time + pyparsing.Literal("-") + _match_time).setParseAction(
-    lambda toks: {
+
+
+def _parse_time_range(toks: typing.List[pyparsing.Token]) -> typing.Any:
+    time1 = toks[0]
+    time2 = toks[2]
+
+    # NOTE(sileht): In case of the format is `10:00-18:00[Europe/Paris]`,
```
Comment (jd) [inline mergify_engine/rules/parser.py:174]: Does this mean you can have:

```
10:00[Europe/Paris]-19:00[US/New York]
```
?

**Review by jd — CHANGES_REQUESTED** [review]

File: mergify_engine/rules/parser.py (hunk 3 above)
Comment (sileht) [reply mergify_engine/rules/parser.py:174]: yes

File: releasenotes/notes/timezone-79b0d5e97fed6bed.yaml (hunk 2 above)
Comment (sileht) [reply releasenotes/notes/timezone-79b0d5e97fed6bed.yaml:3]: I put the explicit list instead.

File: mergify_engine/rules/parser.py (hunk 3 above)
Comment (jd) [reply mergify_engine/rules/parser.py:174]: Maybe write it down in the documentation? 🤔

File: releasenotes/notes/timezone-79b0d5e97fed6bed.yaml [hunk 4]
```diff
@@ -0,0 +1,4 @@
+---
+features:
+  - Allow to specify timezone in ``current-time``, ``schedule``,
+    ``created-at``, ``merged-at``, ``closed-at`` and ``updated_at`` conditions
```
Comment (jd) [inline releasenotes/notes/timezone-79b0d5e97fed6bed.yaml:4]:
```suggestion
    ``created-at``, ``merged-at``, ``closed-at`` and ``updated_at`` conditions.
```

**Review by jd — APPROVED** [review]

**Pull request merged**